src/ui/theme/components/button.ts

```diff
@@ -61,30 +74,52 @@ const variantSolid = (getColor: GetColor) => (
   };
 };
 
-const variantToggle = (getColor: GetColor) => (
+const variantSettings = (getColor: GetColor) => (
   props: React.ComponentProps<typeof Button>
 ) => {
-  const bgColor = getColor('ui.white', 'gray.800', 'ui.white');
-  const color = getColor('gray.800', 'ui.white', 'gray.800');
+  const { bgColor, value } = props;
+
+  const color = getColor('ui.black', 'ui.white', 'ui.black');
+
+  const checkedBgColor = getColor(
+    'ui.gray.light-warm',
+    'ui.gray.x-dark',
+    'ui.sepiaChecked'
+  );
+
   return {
     ...variantSolid(getColor)(props),
-    bgColor,
+    bgColor: getColor('ui.white', 'ui.black', 'ui.sepia'),
+    border: '1px solid',
+    borderBottom: (value === 'paginated' || value === 'scrolling') && 'none',
+    borderRadius:
+      value === 'paginated'
+        ? '0 0 0 4px'
+        : value === 'scrolling'
+        ? '0 0 4px 0'
+        : null,
     color,
-    px: 8,
-    fontWeight: 'medium',
-    fontSize: -2,
-    _focus: {
+    py: 8,
+    width: [8, 16, 36],
+    fontSize: [0, 0, 2],
+    whiteSpace: ['normal', 'normal', 'nowrap'],
+    _active: {
       bgColor,
     },
+    _checked: {
+      color,
+      bgColor: checkedBgColor,
+      borderBottomColor: checkedBgColor,
+      p: {
+        textDecoration: 'underline',
+      },
+    },
     _hover: {
       bgColor,
     },
-    _active: {
+    _focus: {
       bgColor,
-    },
-    _checked: {
-      color: 'ui.white',
-      bgColor: 'green.700',
+      zIndex: 1000,
```

Comment:
Out of curiosity, what was it that required us to set a z index like this when the button is focused? I think this button shouldn't have a z index at all preferably. It should inherit that if it's parent. That way if it is rendered eg in the header, it will have the z index of the header and if it's rendered in a modal it will have the z index of the modal. If we set it explicitly then consider the situation where we render it in the body of the page and then open a modal on top of the page. The modal might have z index of 500 whereas this button has 1000 and therefore is going to float above the modal. Does that make sense?

Reply:
This is the problem I was trying to solve: https://twitter.com/adamwathan/status/1232354577472684033?lang=en

When tabbing, the focus ring is wonky. It only shows on some of the sides of each button because of how the elements overlap. I'm trying to come up with other solutions. Let me know if knowing the issue now makes you think of something better!